vmdb/app/models/miq_request.rb

```diff
@@ -330,16 +332,7 @@ def update_request_status
     req_state = (states.length == 1) ? states.keys.first : "active"
 
     # Determine status to report
-    req_status =
-      if status.keys.include?('Error')
-        'Error'
-      elsif status.keys.include?('Timeout')
-        'Timeout'
-      elsif status.keys.include?('Warn')
-        'Warn'
-      else
-        'Ok'
-      end
+    req_status = status.slice('Error', 'Timeout', 'Warn').keys.first || 'Ok'
```

Comment:
For fun, instead of slice you can also rely on set intersection (which sorts by the left hand side argument)

```ruby
status = {"Timeout" => 1, "Other" => 2, "Error" => 1}
(%w{Error Timeout Warn} & status.keys).first || 'Ok'
# => "Error"
```